cpp/tests/test_utils.h

```diff
@@ -62,6 +62,23 @@ extern "C" {
 }
 #endif
 
+#define NCCLCHECK(cmd) do {                                   \
+    ncclResult_t nccl_status = cmd;                           \
+    if (nccl_status!= ncclSuccess) {                          \
+      printf("NCCL failure %s:%d '%s'\n",                     \
+          __FILE__,__LINE__,ncclGetErrorString(nccl_status)); \
+    }                                                 \
+  } while(0)
+
+#define MPICHECK(cmd) do {                          \
```

Comment:
Minor suggestion about the macro name: `NCCLCHECK` appears to just print messages, but `MPICHECK` calls `exit()`. Maybe this should be called `ENSURE_MPI_SUCCESS` or something more self-documenting, or if you switch it to use a gtest assertion: `ASSERT_MPI_SUCCESS`?

Reply:
Reconciled naming and behavior 0dbc162